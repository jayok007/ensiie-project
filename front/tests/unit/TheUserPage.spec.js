import { mount } from '@vue/test-utils'
import TheUserPage from '@/components/TheUserPage.vue'

describe('TheUserPage.vue', () => {
  it('renders props.msg when passed', async () => {
    const wrapper = mount(TheUserPage, {
      mocks: {
        $api: {
          getUsers () {
            return Promise.resolve(users)
          }
        }
      }
    })

    await wrapper.vm.$nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })
})

const users = [
  {
    id: 1,
    firstname: 'test1',
    lastname: 'test1',
    age: 14
  },
  {
    id: 2,
    firstname: 'test2',
    lastname: 'test2',
    age: 94
  },
  {
    id: 3,
    firstname: 'test3',
    lastname: 'test3',
    age: 54
  }
]
