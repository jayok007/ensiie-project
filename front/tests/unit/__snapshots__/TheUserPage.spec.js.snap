// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheUserPage.vue renders props.msg when passed 1`] = `
<div>
  <div>
    <div class="v-table__overflow">
      <table class="v-datatable v-table theme--light">
        <thead>
          <tr>
            <th role="columnheader" scope="col" aria-label="Id: Sorted ascending. Activate to sort descending." aria-sort="ascending" tabindex="0" class="column sortable active asc text-xs-left">Id<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
            <th role="columnheader" scope="col" aria-label="Firstname: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Firstname<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
            <th role="columnheader" scope="col" aria-label="Lastname: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Lastname<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
            <th role="columnheader" scope="col" aria-label="Age: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Age<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
          </tr>
          <tr class="v-datatable__progress">
            <th colspan="4" class="column"></th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>1</td>
            <td>test1</td>
            <td>test1</td>
            <td>14 years</td>
          </tr>
          <tr>
            <td>2</td>
            <td>test2</td>
            <td>test2</td>
            <td>94 years</td>
          </tr>
          <tr>
            <td>3</td>
            <td>test3</td>
            <td>test3</td>
            <td>54 years</td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
